cartography/data/jobs/analysis/aws_ec2_keypair_analysis.json

```diff
@@ -0,0 +1,30 @@
+{
+    "name": "Analysis jobs for EC2 Key Pairs",
+    "statements": [
+        {
+            "__comment__": "Delete the attribute user_uploaded",
+            "query": "MATCH (k:EC2KeyPair) WHERE EXISTS (k.user_uploaded) REMOVE k.user_uploaded return COUNT(*) as TotalCompleted",
+            "iterative": false
+        },
+        {
+            "__comment__": "Delete the attribute duplicate_keyfingerprint",
+            "query": "MATCH (k:EC2KeyPair) WHERE EXISTS (k.duplicate_keyfingerprint) REMOVE k.duplicate_keyfingerprint return COUNT(*) as TotalCompleted",
+            "iterative": false
+        },
+        {
+            "__comment__": "Delete stale relationships between EC2KeyPairs",
+            "query": "MATCH (k1:EC2KeyPair)-[r:MATCHING_FINGERPRINT]->(k2:EC2KeyPair) WHERE r.lastupdated <> {UPDATE_TAG} DELETE (r) return COUNT(*) as TotalCompleted",
+            "iterative": false
+        },
+        {
+            "__comment__": "Set EC2KeyPairs with MD5 hash keyfingerprints (47 characters, instead of 59 characters) as user_uploaded = True",
```

Comment:
Non-blocking question: I'm just curious, but do you know of any AWS docs that talk about these lengths?  If so it would be helpful to include in the schema.

Reply:
Can't find it. But I used this wiki article as reference for the digest size in bits, then figuring the character length in hex and colons. 
https://en.m.wikipedia.org/wiki/Cryptographic_hash_function#Cryptographic_hash_algorithms